src/hardhat.ts

```diff
@@ -0,0 +1,68 @@
+import { CHAINS } from './generated/chains';
+import { toUpperSnakeCase } from './utils/strings';
+import { Chain, HardhatEtherscanConfig, HardhatNetworksConfig } from './types';
+
+export function buildEnvVariables(): string[] {
+  return CHAINS
+    .filter((chain) => chain.explorer?.api?.key?.required)
+    .map((chain) => buildEtherscanApiKeyName(chain));
+}
+
+export function buildEtherscanApiKeyName(chain: Chain): string {
+  return `${toUpperSnakeCase(chain.alias)}_ETHERSCAN_API_KEY`;
+}
+
+// https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-verify#multiple-api-keys-and-alternative-block-explorers
+export function buildEtherscanConfig(): HardhatEtherscanConfig {
+  // Not usable outside of a Node.js environment
+  if (typeof window !== 'undefined') {
+    return { apiKey: {}, customChains: [] };
+  }
+
+  return CHAINS.reduce((etherscan, chain) => {
+    if (!chain.explorer || !chain.explorer.api) {
+      return etherscan;
+    }
+
+    const explorer = chain.explorer;
+    const apiKey = chain.explorer.api.key;
+
+    const apiKeyEnvName = buildEtherscanApiKeyName(chain);
+    const apiKeyValue = apiKey.required ? chain.alias : process.env[apiKeyEnvName];
```

Comment:
@bbenligiray I don't know if this line is correct still or not. I suspect it should maybe be this instead?

```suggestion
    const apiKeyValue = apiKey.required ? process.env[apiKeyEnvName] : chain.alias;
```

Reply:
Yes, it should be that instead, though I'd prefer
```
    const apiKeyValue = apiKey.required ? process.env[apiKeyEnvName] : 'DUMMY_VALUE';
```
This comes off as if the dummy value being the alias has a significance